jobserver/models.py

```diff
@@ -64,13 +65,28 @@ def repo_name(self):
 
 class JobQuerySet(models.QuerySet):
     def completed(self):
-        return self.filter(completed_at__isnull=False)
-
-    def in_progress(self):
-        return self.filter(started_at__isnull=False, completed_at=None)
+        return (
+            self.filter(started=True)
+            .exclude(completed_at=None)
+            .exclude(status_code=None)
+        )
+
+    def finished(self):
```

Comment:
Not understanding the intent of differentiating between `finished` and `completed` here.

Also, the logic for `completed` makes sense but the logic for `finished` would include things that are pending, no?

(As an aside, personally I find the use of `exclude` confusing and would find `filter` with negative assertions like `completed_at__isnull` easier to parse)

Reply:
* `finished`: the job has run and is no longer running
* `completed`: the job has finished with a success mode
* `failed`: the job has finished with a failure mode

I'm thinking `succeeded` would be the better name here?  It's a stronger signal to the positive nature of the state?

We need the `finished` state to make aggregating state at the JobRequest level much easier.

---

> the logic for finished would include things that are pending

I don't think so?  It's checking a Job has been started and has `completed_at` set.  While pending short-circuits when `started=False`.

---

I often flip flop on `.exclude()`, `__isnull` can be hard to parse for some folks (including me at times) so I reach for `.exclude()` when I think it's going to be tricky to read those.  I'll see how `__isnull` works for these.

Reply:
I prefer `succeeded` to `completed`.

Reply:
Switched all the JobQuerySet methods over to the `__isnull` style.  I'm dubious on the readabily improvements but I did find a bug in `.finished()` 😬